### DISCUSSION TITLE
How hard is it to add support for local transport, such as named pipe or domain socket?

### DISCUSSION THREAD

**Discussion by linw789** [microsoft/msquic#3430] [General]
I haven't looked closely at msquic. I saw it was mentioned in another thread that msquic had to own socket itself due to complicated settings it required. I wonder how hard, if not impossible, it would be to add support for local transport such as named pipe on Windows or domain socket on Linux. I'm exploring ideas of using msquic to manage both remote, as well as inter-process communication.

**Comment by nibanks** [comment]
Sorry I'm just getting to this. Our data path abstraction has already been implemented 5 or 6 times, so you could make another one. I wouldn't say it's _easy_ though.
